File: CS/DxSample/Grid/GridViewEx.cs
using DevExpress.Data;
using DevExpress.Utils.Controls;
using DevExpress.Utils.Serializing;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid;
using System;
using System.ComponentModel;

namespace DxSample.Grid {
    public class GridViewEx :GridView {
        public GridViewEx() : base() { }
        public GridViewEx(GridControl ownerGrid) : base(ownerGrid) { }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content), XtraSerializableProperty(XtraSerializationVisibility.Content)]
        public new GridViewExOptionsBehavior OptionsBehavior {
            get { return (GridViewExOptionsBehavior)base.OptionsBehavior; }
        }

        protected override ColumnViewOptionsBehavior CreateOptionsBehavior() {
            return new GridViewExOptionsBehavior(this);
        }

        protected override BaseGridController CreateDataController() {
            if (requireDataControllerType == DataControllerType.Regular)
                return new CancellingChangesDataController();
            return base.CreateDataController();
        }

        protected override string OnCreateLookupDisplayFilter(string text, string displayMember) {
            throw new NotSupportedException("This view cannot be used in lookup editors");
        }

        protected override void UpdateDataControllerOptions() {
            base.UpdateDataControllerOptions();
            CancellingChangesDataController controller = DataController as CancellingChangesDataController;
            if (controller != null)
                controller.CanCancelChanges = OptionsBehavior.CanCancelChanges;
        }
    }
}

File: VB/DxSample/Grid/GridViewEx.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Data
Imports DevExpress.Utils.Controls
Imports DevExpress.Utils.Serializing
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid
Imports System
Imports System.ComponentModel

Namespace DxSample.Grid
	Public Class GridViewEx
		Inherits GridView
		Public Sub New()
			MyBase.New()
		End Sub
		Public Sub New(ByVal ownerGrid As GridControl)
			MyBase.New(ownerGrid)
		End Sub

		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content), XtraSerializableProperty(XtraSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property OptionsBehavior() As GridViewExOptionsBehavior
			Get
				Return CType(MyBase.OptionsBehavior, GridViewExOptionsBehavior)
			End Get
		End Property

		Protected Overrides Function CreateOptionsBehavior() As ColumnViewOptionsBehavior
			Return New GridViewExOptionsBehavior(Me)
		End Function

		Protected Overrides Function CreateDataController() As BaseGridController
			If requireDataControllerType = DataControllerType.Regular Then
				Return New CancellingChangesDataController()
			End If
			Return MyBase.CreateDataController()
		End Function

		Protected Overrides Function OnCreateLookupDisplayFilter(ByVal text As String, ByVal displayMember As String) As String
			Throw New NotSupportedException("This view cannot be used in lookup editors")
		End Function

		Protected Overrides Sub UpdateDataControllerOptions()
			MyBase.UpdateDataControllerOptions()
			Dim controller As CancellingChangesDataController = TryCast(DataController, CancellingChangesDataController)
			If controller IsNot Nothing Then
				controller.CanCancelChanges = OptionsBehavior.CanCancelChanges
			End If
		End Sub
	End Class
End Namespace
